src/main/kotlin/com/sourcegraph/cody/auth/CodyAccountService.kt

```diff
@@ -0,0 +1,56 @@
+package com.sourcegraph.cody.auth
+
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import com.sourcegraph.cody.auth.CodySecureStore.getFromSecureStore
+import com.sourcegraph.cody.auth.CodySecureStore.writeToSecureStore
+
+@Service(Service.Level.PROJECT)
+class CodyAccountService(val project: Project) {
+  private val activeAccountMarker = "active_cody_account"
+  private val projectActiveAccountMarker = project.locationHash
+  private var activeAccount: CodyAccount? = null
+
+  @Volatile private var isActivated: Boolean = false
+
+  fun isActivated(): Boolean {
+    return isActivated
+  }
+
+  fun setActivated(isActivated: Boolean) {
+    this.isActivated = isActivated
+  }
+
+  fun loadAccount(serverUrl: String): CodyAccount {
+    val token = serverUrl.let { getFromSecureStore(it) }
+    return CodyAccount(SourcegraphServerPath(serverUrl), token)
+  }
+
+  fun storeAccount(account: CodyAccount) {
+    writeToSecureStore(account.server.url, account.token)
+  }
+
+  fun getActiveAccount(): CodyAccount? {
```

Comment:
When we do JetBrains-specific state that's not coordinated with TypeScript/agent, we usually end up regretting it.

What happens if the project active account marker points to a signed out account?